### ISSUE TITLE
[BUG]: Corrupted Reverse Geocoding CSV File 

### ISSUE THREAD

**Issue by jrasm91** [immich-app/immich#1963]
https://github.com/tomayac/local-reverse-geocoder/issues/63

Possible solutions:
- Fix upstream
- Detect the issue, delete the csv file/directory, and try again

**Comment by twitsforbrains** [comment]
FYI I created a pull request in the upstream for this.

**Comment by MMauro94** [comment]
Hello, just wanted to share that this keeps failing for me. Even after recreating the microservices container I get this error shortly after starting it:
```
[Nest] 1  - 04/06/2023, 7:10:41 PM     LOG [MediaService] Start encoding video 417b1bc3-471c-4533-869b-98c00d9f35bb -crf 23,-preset ultrafast,-vcodec h264,-acodec aac,-movflags faststart,-vf scale=-2:720
[Nest] 1  - 04/06/2023, 7:10:41 PM     LOG [MediaService] Start encoding video a8f851a9-e466-498a-bb44-5f597891b029 -crf 23,-preset ultrafast,-vcodec h264,-acodec aac,-movflags faststart,-vf scale=-2:720
/usr/src/app/node_modules/local-reverse-geocoder/index.js:746
            throw err;
            ^
Error downloading GeoNames admin 2 codes data: Error: read ECONNRESET
(Use `node --trace-uncaught ...` to show where the exception was thrown)
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [NestFactory] Starting Nest application...
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +213ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] BullModule dependencies initialized +1ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] ConfigHostModule dependencies initialized +3ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] DiscoveryModule dependencies initialized +1ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] ConfigModule dependencies initialized +23ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] BullModule dependencies initialized +2ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] BullModule dependencies initialized +1ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] TypeOrmCoreModule dependencies initialized +543ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +1ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +1ms
[Nest] 1  - 04/06/2023, 7:12:29 PM    WARN [MetadataExtractionProcessor] Reverse geocoding is enabled
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [MetadataExtractionProcessor] Initializing Reverse Geocoding
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] InfraModule dependencies initialized +18ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] DomainModule dependencies initialized +1ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [InstanceLoader] MicroservicesModule dependencies initialized +1ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [NestApplication] Nest application successfully started +49ms
[Nest] 1  - 04/06/2023, 7:12:29 PM     LOG [ImmichMicroservice] Running Immich Microservices in PRODUCTION environment - version 1.53.0 - Listening on port: 3002
/usr/src/app/node_modules/local-reverse-geocoder/index.js:746
            throw err;
            ^
CsvError: Invalid Record Length: expect 19, got 14 on line 10326
    at Object.__onRecord (/usr/src/app/node_modules/csv-parse/dist/cjs/index.cjs:940:11)
    at Object.parse (/usr/src/app/node_modules/csv-parse/dist/cjs/index.cjs:903:36)
    at Parser._flush (/usr/src/app/node_modules/csv-parse/dist/cjs/index.cjs:1336:26)
    at Parser.final [as _final] (node:internal/streams/transform:112:25)
    at callFinal (node:internal/streams/writable:694:27)
    at prefinish (node:internal/streams/writable:723:7)
    at finishMaybe (node:internal/streams/writable:733:5)
    at afterWrite (node:internal/streams/writable:504:3)
    at onwrite (node:internal/streams/writable:477:7)
    at Parser.Transform._read (node:internal/streams/transform:245:5) {
  code: 'CSV_RECORD_INCONSISTENT_FIELDS_LENGTH',
  bytes: 1589248,
  comment_lines: 0,
  empty_lines: 0,
  invalid_field_length: 0,
  lines: 10326,
  records: 10325,
  columns: false,
  error: undefined,
  header: false,

  index: 14,
  raw: undefined,
  column: 14,
  quoting: false,
  record: [
    '2801970',  'Biesme',
    'Biesme',   'Mettet (Biesme)',
    '50.34015', '4.61048',
    'P',        'PPL',
    'BE',       '',
    'WAL',      'WNA',
    '92',       '92'
  ]
}
```

This has started happening after performing a batch upload with the CLI of all my media from 2021. I am now left in a state where the media is uploaded (it's in the `upload/{USER-UID}` directory) but is unable to be processed.

I tried getting the `cities500.zip` from inside the container and it seems to work fine:
```
/usr/src/app # wget https://download.geonames.org/export/dump/cities500.zip
Connecting to download.geonames.org (5.9.152.54:443)
saving to 'cities500.zip'
cities500.zip        100% |**********************************************************************************************************************************************| 10.0M  0:00:00 ETA
'cities500.zip' saved
/usr/src/app #
```

I don't have any geocoding-related env variables, so I'm assuming it's using as default `REVERSE_GEOCODING_PRECISION=3`.

I've recreated the container endless times now but the result is always the same.

**Comment by skyzem** [comment]
> Hello, just wanted to share that this keeps failing for me. Even after recreating the microservices container I get this error shortly after starting it:

same for me! I also did a CLI batch upload of various years.

`[Nest] 1  - 05/03/2023, 4:44:22 PM     LOG [NestFactory] Starting Nest application...
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +211ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] BullModule dependencies initialized +2ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] ConfigHostModule dependencies initialized +3ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] DiscoveryModule dependencies initialized +1ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] ConfigModule dependencies initialized +26ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] BullModule dependencies initialized +2ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] BullModule dependencies initialized +1ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] TypeOrmCoreModule dependencies initialized +422ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +2ms
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +1ms
[Nest] 1  - 05/03/2023, 4:44:23 PM    WARN [MetadataExtractionProcessor] Reverse geocoding is enabled
[Nest] 1  - 05/03/2023, 4:44:23 PM     LOG [MetadataExtractionProcessor] Initializing Reverse Geocoding
/usr/src/app/node_modules/local-reverse-geocoder/index.js:746
            throw err;
            ^
CsvError: Invalid Record Length: expect 19, got 15 on line 179072
    at Object.__onRecord (/usr/src/app/node_modules/csv-parse/dist/cjs/index.cjs:940:11)
    at Object.parse (/usr/src/app/node_modules/csv-parse/dist/cjs/index.cjs:903:36)
    at Parser._flush (/usr/src/app/node_modules/csv-parse/dist/cjs/index.cjs:1336:26)
    at Parser.final [as _final] (node:internal/streams/transform:112:25)
    at callFinal (node:internal/streams/writable:694:27)
    at prefinish (node:internal/streams/writable:723:7)
    at finishMaybe (node:internal/streams/writable:733:5)
    at afterWrite (node:internal/streams/writable:504:3)
    at onwrite (node:internal/streams/writable:477:7)
    at Parser.Transform._read (node:internal/streams/transform:245:5) {
  code: 'CSV_RECORD_INCONSISTENT_FIELDS_LENGTH',
  bytes: 28737536,
  comment_lines: 0,
  empty_lines: 0,
  invalid_field_length: 0,
  lines: 179072,
  records: 179071,
  columns: false,
  error: undefined,
  header: false,
  index: 15,
  raw: undefined,
  column: 15,
  quoting: false,
  record: [
    '4295878',
    'Inez',
    'Inez',
    'Ajnez,Eden,Inez,anyz,aynz  kntaky,yi nei ci,Айнез,Ајнез,إنيز,اینز، کنتاکی,伊内兹',
    '37.86648',
    '-82.53876',
    'P',
    'PPLA2',
    'US',
    '',
    'KY',
    '159',
    '',
    '',
    '6'
  ]
}`

**Comment by sakowicz** [comment]
This is also a case in https://github.com/immich-app/immich/issues/2367 . 
Can we do something to handle this? Is the issue reported in the upstream repo? Does someone understand the reason for this bug?

**Comment by alextran1502** [comment]
@sakowicz The plan is to fork and rewrite the library to properly handle error so it does not crash

**Comment by ceebu** [comment]
Is there a way to prevent immich from trying to Geocode ? I am struggling with this issue and my uploads of old photos always stops half way. I have tried uploading 3 or 4 times now and would be great if I can run immich without tagging location.

**Comment by jrasm91** [comment]
You can disable it with an env.

```
DISABLE_REVERSE_GEOCODING=true
```

**Comment by wavesswe** [comment]
Same here did "DISABLE_REVERSE_GEOCODING=true" for now and it seams to be a work around.

**Comment by jrasm91** [comment]
Another attempted fix with #2489

**Comment by jrasm91** [comment]
I think this has been fixed. We can reopen if that is not the case.
